twistlock/datadog_checks/twistlock/twistlock.py

```diff
@@ -0,0 +1,322 @@
+    "high": ["severity:high", "min_severity:low", "min_severity:medium", "min_severity:high"],
+    "critical": ["severity:critical", "min_severity:low", "min_severity:medium",
+                 "min_severity:high", "min_severity:critical"]
+}
+
+
+class TwistlockCheck(AgentCheck):
+    NAMESPACE = 'twistlock'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+        msg = 'This check has not run before, '
+        msg += 'it will send all the new vulnerabilities that happened in the past day'
+        self.log.debug(msg)
+        self.last_run = datetime.now() - timedelta(days=1)
+
+    def check(self, instance):
+        if 'url' not in instance:
+            raise Exception('Instance missing "url" value.')
+
+        self.config = Config(instance)
+
+        self.current_date = datetime.now()
+        self.warning_date = self.current_date - timedelta(hours=7)
+        self.critical_date = self.current_date - timedelta(days=1)
+
+        self.report_license_expiration()
+        self.report_registry_scan()
+        self.report_images_scan()
+        self.report_hosts_scan()
+        self.report_container_compliance()
+
+        self.report_vulnerabilities()
+
+        self.last_run = datetime.now()
+
+    def report_license_expiration(self):
+        service_check_name = self.NAMESPACE + ".license_ok"
+        try:
+            license = self._retrieve_json("/api/v1/settings/license")
+        except Exception as e:
+            self.warning("cannot retrieve license data: {}".format(e))
+            self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
+            raise e
+
+        expiration_date = datetime.strptime(license.get("expiration_date"), LICENSE_DATE_FORMAT)
+        current_date = datetime.now()
+        warning_date = current_date + timedelta(days=30)
+        critical_date = current_date + timedelta(days=7)
+
+        licence_status = AgentCheck.OK
+        if expiration_date < warning_date:
+            licence_status = AgentCheck.WARNING
+        if expiration_date < critical_date:
+            licence_status = AgentCheck.CRITICAL
+        self.service_check(service_check_name, licence_status,
+                           tags=self.config.tags, message=license.get("expiration_date"))
+
+    def report_registry_scan(self):
+        namespace = self.NAMESPACE + ".registry"
+        service_check_name = self.NAMESPACE + ".can_connect"
+        try:
+            scan_result = self._retrieve_json("/api/v1/registry")
+            self.service_check(service_check_name, AgentCheck.OK, tags=self.config.tags)
+        except Exception as e:
+            self.warning("cannot retrieve registry data: {}".format(e))
+            self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
+            return None
+
+        for image in scan_result:
+            if '_id' not in image:
```

Comment:
What is this for?

Reply:
it skips over things that aren't valid objects